### DISCUSSION TITLE
Upgrade after webhook instead of polling

### DISCUSSION THREAD

**Discussion by lonix1** [containrrr/watchtower#1709] [Q&A] [answered]
New user here. I've read the docs.

I'd like to use watchtower to listen for webhooks from my cicd server (woodpecker) and only then upgrade a **specific** container to the latest version.

The docs don't mention "webhooks" anywhere, but the Arguments page mentions an "HTTP API mode".

Am I right in assuming that I can use that to receive webhooks? Are there example compose files anywhere which show how to configure that?

Thank you.

**Comment by piksel** [comment, marked as answer]
Yeah, so, the API can absolutely be used that way, if you want to expose the endpoint to the web without having to provide an API key you can use nginx or a similar proxy that adds the key to the request. 
I don't have a ready made compose file for this, but what you need is:
1. Enable Update API
2. Provide an API key (can be anything you want)
3. Add the container name to the command
4. (optionally) Proxy a request to watchtower from the web

You can read more here:
https://containrrr.dev/watchtower/http-api-mode/

(including an example compose file)

**Reply by lonix1** [reply]
Thanks! I don't know why I didn't see that document.

**Reply by lonix1** [reply]
Oh now it makes sense - it's [in the code](https://github.com/containrrr/watchtower/blob/main/mkdocs.yml#L51), but is not shown in the docs site. So I missed it.
